.github/workflows/ci.yml

```diff
@@ -0,0 +1,30 @@
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+name: CI
+
+jobs:
+  test:
+    name: run tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version: '^1.13.0'
+      - name: Run tests
+        run: go test
+
+  fmt:
+    name: check code formatting with gofmt
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version: '^1.13.0'
+      - name: run gofmt
+        run: diff -u <(echo -n) <(gofmt -d ./)
```

Comment:
Isn't it enough to just run `gofmt` with the `-d` flag?

Reply:
Nah, for some reason they seem to think that it's helpful to return exit code 0 even if there are changes...

https://stackoverflow.com/questions/42510140/check-format-for-continous-integration

Reply:
I mean I'd expect it to have a `--check` flag or similar and then return an error, but it doesn't seem to have that.

Reply:
I expected such a flag as well, but it probably breaks some kind of perceived unix philosophy - why implement a feature if you can do it with convoluted shell pipelines as well? 😂